controllers/machinedeployment_sync.go

```diff
@@ -113,8 +113,14 @@ func (r *MachineDeploymentReconciler) getNewMachineSet(ctx context.Context, d *c
 		annotationsUpdated := mdutil.SetNewMachineSetAnnotations(d, msCopy, newRevision, true, log)
 
 		minReadySecondsNeedsUpdate := msCopy.Spec.MinReadySeconds != *d.Spec.MinReadySeconds
-		if annotationsUpdated || minReadySecondsNeedsUpdate {
+		deletePolicyNeedsUpdate := d.Spec.Strategy.RollingUpdate.DeletePolicy != nil && msCopy.Spec.DeletePolicy != *d.Spec.Strategy.RollingUpdate.DeletePolicy
+		if annotationsUpdated || minReadySecondsNeedsUpdate || deletePolicyNeedsUpdate {
 			msCopy.Spec.MinReadySeconds = *d.Spec.MinReadySeconds
+
+			if d.Spec.Strategy.RollingUpdate.DeletePolicy != nil {
```

Comment:
Can we reuse `if deletePolicyNeedsUpdate` here?

Reply:
Tried to save a comparison check :) yeah, I'll change it.